### ISSUE TITLE
Missing index entries when using bleve with moss

### ISSUE THREAD

**Issue by ck-schmidi** [blevesearch/bleve#614]
We switched from boltdb to moss for the underlying K/V store, because the performance for indexing 
was much better. After a while we noticed, that in some cases, there were missing entries on the index. 
So i wrote a little reproducable test programm which demonstrates this problem. 

When `index.Close()` gets called, somehow the indexing of the remaining items gets canceled. If we run the test and wait with `time.Sleep(1 * time.Second)` before `index.Close()` gets called, all items are successfully saved to the index. 

The same tests using `boltdb` works fine!

testprogram: [link to gist](https://gist.github.com/ck-schmidi/ad4123d291d66e45b77ec524bdc5b693)

**Comment by mschoch** [comment]
Yes, moss uses asynchronous persistence.  This is expected behavior, but it's insufficiently documented.

I'm going to propose we move moss out of bleve into blevex package since it does not honor the bleve interface and is causing confusion.

**Comment by ck-schmidi** [comment]
Yes, I totally get the asynchronous persistence mechanism. 
But why not waiting for all persistence jobs, on `index.Close()`?

**Comment by mschoch** [comment]
Well, there are several use cases to support.  In our case, we never want to wait for persistence, even on close because we can always resume later, and Close() might take several hours to return depending on how much data is not yet persisted.

Sure, we could change the defaults, so that it works slightly better out of the box.  But, my fear now is that just fixing Close() simply hides the true problems deeper.  Today, you find out relatively quickly that moss doesn't work as expected.  If Close() were to wait for persistence, you'd find that moss works correctly in more cases.  But, if your process ever crashes unexpectedly, you'll again be disappointed with moss having lost your data.

The only way to safely use moss today is with a datasource that is completely resumable.  In these cases, losing data is always OK, as you can always just redo what you lost.
